File: VB/GridCellsCreepingLine/Form1.vb
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace GridCellsCreepingLine
    Partial Public Class Form1
        Inherits Form

        Private column As GridColumn = Nothing
        Private creep As Boolean = False
        Private rowHandle As Integer = -1

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.DataSource = GetPersonDataTable()
            AddHandler gridControl1.GetToolTipController().GetActiveObjectInfo, New ToolTipControllerGetActiveObjectInfoEventHandler(AddressOf OnGridControlToolTipControllerGetActiveObjectInfo)
        End Sub

        Private Function GetPersonDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "Persons"
            table.Columns.Add(New DataColumn("FirstName", GetType(String)))
            table.Columns.Add(New DataColumn("SecondName", GetType(String)))
            table.Columns.Add(New DataColumn("Age", GetType(Integer)))
            table.Columns.Add(New DataColumn("ID", GetType(Integer)))

            For i As Integer = 0 To 50 - 1
                table.Rows.Add("Adam van Dorian " & i, "Peterson-Don-Abraham " & i, 20 + i / 2, i)
            Next

            Return table
        End Function

        Private Sub gridView1_CustomDrawCell(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs)
            If creep AndAlso e.Column Is column AndAlso e.RowHandle = rowHandle Then
                Dim szCellValue As SizeF = e.Appearance.CalcTextSize(e.Graphics, Convert.ToString(e.CellValue) & ".", 1000)

                If szCellValue.Width > e.Bounds.Width Then
                    If e.DisplayText = Convert.ToString(e.CellValue) Then e.DisplayText = e.DisplayText + " "c
                    Dim letter As Char = e.DisplayText(0)
                    e.DisplayText = e.DisplayText.Remove(0, 1)
                    e.DisplayText = e.DisplayText + letter
                End If
            Else
                e.DisplayText = e.DisplayText
            End If
        End Sub

        Private Sub gridView1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim view As GridView = TryCast(sender, GridView)
            Dim hitInfo As GridHitInfo = view.CalcHitInfo(e.X, e.Y)

            If hitInfo.HitTest = GridHitTest.RowCell Then

                If column IsNot hitInfo.Column OrElse rowHandle <> hitInfo.RowHandle Then
                    Dim previousRowhandle As Integer = rowHandle
                    Dim previousColumn As GridColumn = column
                    column = hitInfo.Column
                    rowHandle = hitInfo.RowHandle
                    view.RefreshRowCell(previousRowhandle, previousColumn)
                End If
            End If
        End Sub

        Private Sub OnGridControlToolTipControllerGetActiveObjectInfo(ByVal sender As Object, ByVal e As ToolTipControllerGetActiveObjectInfoEventArgs)
            If e.Info Is Nothing Then Return
            Dim hitInfo As GridHitInfo = gridView1.CalcHitInfo(e.ControlMousePosition.X, e.ControlMousePosition.Y)
            If hitInfo.HitTest = GridHitTest.RowCell Then e.Info.Text = Convert.ToString(gridView1.GetRowCellValue(hitInfo.RowHandle, hitInfo.Column))
        End Sub

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs)
            creep = True
            gridView1.InvalidateRowCell(rowHandle, column)
        End Sub
    End Class
End Namespace

File: CS/GridCellsCreepingLine/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace GridCellsCreepingLine
{
    public partial class Form1 : Form
    {
        private GridColumn column = null;
        private bool creep = false;
        private int rowHandle = -1;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = GetPersonDataTable();
            gridControl1.GetToolTipController().GetActiveObjectInfo += new ToolTipControllerGetActiveObjectInfoEventHandler(OnGridControlToolTipControllerGetActiveObjectInfo);
        }
        DataTable GetPersonDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Persons";
            table.Columns.Add(new DataColumn("FirstName", typeof(string)));
            table.Columns.Add(new DataColumn("SecondName", typeof(string)));
            table.Columns.Add(new DataColumn("Age", typeof(int)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for(int i = 0; i < 50; i++)
                table.Rows.Add("Adam van Dorian " + i, "Peterson-Don-Abraham " + i, 20 + i / 2, i);
            return table;
        }

        private void gridView1_CustomDrawCell(object sender, DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs e)
        {
            if(creep && e.Column == column && e.RowHandle == rowHandle)
            {
                SizeF szCellValue = e.Appearance.CalcTextSize(e.Graphics, Convert.ToString(e.CellValue) + ".", 1000);
                if(szCellValue.Width > e.Bounds.Width)
                {
                    if(e.DisplayText == Convert.ToString(e.CellValue))
                        e.DisplayText = e.DisplayText + ' ';

                    char letter = e.DisplayText[0];
                    e.DisplayText = e.DisplayText.Remove(0, 1);
                    e.DisplayText = e.DisplayText + letter;
                }
            }
            else { e.DisplayText = e.DisplayText; }
        }

        private void gridView1_MouseMove(object sender, MouseEventArgs e)
        {
            GridView view = sender as GridView;
            GridHitInfo hitInfo = view.CalcHitInfo(e.X, e.Y);
            if(hitInfo.HitTest == GridHitTest.RowCell)
            {
                if(column != hitInfo.Column || rowHandle != hitInfo.RowHandle)
                {
                    int previousRowhandle = rowHandle;
                    GridColumn previousColumn = column;
                    column = hitInfo.Column;
                    rowHandle = hitInfo.RowHandle;
                    view.RefreshRowCell(previousRowhandle, previousColumn);
                }
            }
        }
        private void OnGridControlToolTipControllerGetActiveObjectInfo(object sender, ToolTipControllerGetActiveObjectInfoEventArgs e)
        {
            if(e.Info == null)
                return;

            GridHitInfo hitInfo = gridView1.CalcHitInfo(e.ControlMousePosition.X, e.ControlMousePosition.Y);
            if(hitInfo.HitTest == GridHitTest.RowCell)
                e.Info.Text = Convert.ToString(gridView1.GetRowCellValue(hitInfo.RowHandle, hitInfo.Column));
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            creep = true;
            gridView1.InvalidateRowCell(rowHandle, column);
        }
    }
}